api/organisations/views.py

```diff
@@ -272,6 +355,12 @@ def chargebee_webhook(request):
      - If subscription is cancelled or not renewing, update subscription on our end to include cancellation date and
        send alert to admin users.
     """
+    event_type = request.data.get("event_type")
+
+    if event_type == "payment_failed":
+        return chargebee_webhook_payment_failed(request)
+    if event_type == "payment_succeeded":
+        return chargebee_webhook_payment_succeeded(request)
```

Comment:
I'm not 100% sure we want to early exit here. The way this function operated before I added the payment failed and payment succeeded early exits, rightly or wrongly, was to look at every single webhook and if it had subscription data, update the subscription information we have in the data base. This shouldn't be necessary since only the subscription specific webhooks should be necessary for keeping our subscription information up to date, but if we want to handle the payment_failed and payment_succeeded requests in the same way it is a simple change.

Reply:
I think this is fine. As per the above though, I'd love to start using some level of serialization here as well.